src/v/model/metadata.h

```diff
@@ -424,6 +424,16 @@ operator<<(std::ostream& o, const partition_autobalancing_mode& m) {
     }
 }
 
+enum class cloud_storage_backend {
+    aws = 0,
+    google = 1,
```

Comment:
Let's define the google one as "google_xml" or similar, in case we ever add support for the native JSON GCS API.

Reply:
I renamed it to `google_s3_compat` to convey the intent.